tardis/visualization/tools/rpacket_visualization.py

```diff
@@ -0,0 +1,686 @@
+            (
+                rpacket_x,
+                rpacket_y,
+                rpacket_interactions,
+            ) = self.get_coordinates_with_theta_init(
+                r_packet_tracker.loc[packet_no]["r"],
+                r_packet_tracker.loc[packet_no]["mu"],
+                self.sim.model.time_explosion.value,
+                r_packet_tracker.loc[packet_no]["interaction_type"],
+                thetas[packet_no],
+            )
+            rpackets_x.append(rpacket_x)
+            rpackets_y.append(rpacket_y)
+            rpackets_interactions.append(rpacket_interactions)
+        return (
+            np.array(rpackets_x, dtype="object"),
+            np.array(rpackets_y, dtype="object"),
+            np.array(rpackets_interactions, dtype="object"),
+        )
+
+    def get_equal_array_size(self, rpacket_x, rpacket_y, interactions):
+        """
+        creates the coordinate arrays of different packets of same size. This is done for generating frames in animation.
+
+        Parameters
+        ----------
+        rpacket_x : numpy.ndarray
+            x coordinates of packets
+        rpacket_y : numpy.ndarray
+            y coordinates of packets
+        interactions : numpy.ndarray
+            interaction types of packets
+
+        Returns
+        -------
+        numpy.ndarray
+            normalized x coordinate array
+        numpy.ndarray
+            normalized y coordinate array
+        numpy.ndarray
+            normalized interaction types array
+        int
+            size of the biggest array among different packets
+
+        """
+        rpacket_step_no_array_max_size = max(list(map(len, rpacket_x)))
+
+        for packet_no in range(len(rpacket_x)):
+            # making all coordinate arrays of size `rpacket_step_no_array_max_size` by repeating the last element across the remaining length of array
+            rpacket_x[packet_no] = np.append(
+                rpacket_x[packet_no],
+                rpacket_x[packet_no][-1]
+                * np.ones(
+                    [rpacket_step_no_array_max_size - len(rpacket_x[packet_no])]
+                ),
+            )
+            rpacket_y[packet_no] = np.append(
+                rpacket_y[packet_no],
+                rpacket_y[packet_no][-1]
+                * np.ones(
+                    [rpacket_step_no_array_max_size - len(rpacket_y[packet_no])]
+                ),
+            )
+            interactions[packet_no] = np.append(
+                interactions[packet_no],
+                interactions[packet_no][-1]
+                * np.ones(
+                    [
+                        rpacket_step_no_array_max_size
+                        - len(interactions[packet_no])
+                    ]
+                ),
+            )
+        return (
+            rpacket_x,
+            rpacket_y,
+            interactions,
+            rpacket_step_no_array_max_size,
+        )
+
+    # creating frames for animation
```

Comment:
I don't think you need comments outside of functions like this. This comment is basically the same as what's in the docstring. It would be more useful to have some comments inside the for loop explaining what's being done there instead.